main.nf

```diff
@@ -0,0 +1,932 @@
+    """    
+    }
+}
+
+
+/*
+ * Step 9 - Genotype generate GVCFs using GATK's GenotypeGVCFs
+ * 
+*/ 
+
+process genotypegvcfs{
+    tag "${name}"
+
+    input:
+    set val(name), file(raw_vcf), file(raw_vcf_idx) from raw_variants
+
+    output:
+    set val(name), file("${name}_gvcf.vcf"), file("${name}_gvcf.vcf.idx") into raw_gvcfs
+
+    script:
+    """
+    gatk -T GenotypeGVCFs \\
+    -R $params.gfasta \\
+    --variant $raw_vcf \\
+    -nt $task.cpus \\
+    -o ${name}_gvcf.vcf \\
+    """
+}
+
+/*
+ * Step 10 - Create separate files for SNPs and Indels 
+ * 
+*/ 
+
+process variantSelect {
+    tag "${name}"
+
+    input:
+    set val(name), file(raw_vcf), file(raw_vcf_idx) from raw_gvcfs
+
+    output:
+    set val(name), file("${name}_snp.vcf"), file("${name}_snp.vcf.idx") into raw_snp
+    set val(name), file("${name}_indels.vcf"), file("${name}_indels.vcf.idx") into raw_indels
+
+    script:
+    """
+    gatk -T SelectVariants \\
+        -R $params.gfasta \\
+        --variant $raw_vcf \\
+        --out ${name}_snp.vcf \\
+        --selectTypeToInclude SNP
+
+    gatk -T SelectVariants \\
+        -R $params.gfasta \\
+        --variant $raw_vcf \\
+        --out ${name}_indels.vcf \\
+        --selectTypeToInclude INDEL \\
+        --selectTypeToInclude MIXED \\
+        --selectTypeToInclude MNP \\
+        --selectTypeToInclude SYMBOLIC \\
+        --selectTypeToInclude NO_VARIATION
+    """
+}
+
+
+/*
+ * Step 11 - Recalibrate SNPs using Omni, 1000G and DBSNP databases 
+ * 
+*/ 
+
+process recalSNPs {
+    tag "${name}"
+    publishDir "${params.outdir}/variantCallingRecalibrated/", mode: 'copy'
+
+    input:
+    set val(name), file(raw_snp), file(raw_snp_idx) from raw_snp
+
+    output:
+    set val(name), file("${sample}_filtered_snp.vcf"), file("${sample}_filtered_snp.vcf.idx") into filtered_snp
+
+    script:
+    """
+    gatk -T VariantRecalibrator \\
+        -R $params.gfasta \\
+        --input $raw_snp \\
+        --maxGaussians 4 \\
+        --recal_file ${name}_snp.recal \\
+        --tranches_file ${name}_snp.tranches \\
+        -resource:omni,known=false,training=true,truth=true,prior=12.0 $params.omni \\
+        -resource:1000G,known=false,training=true,truth=false,prior=10.0 $params.thousandg \\
+        -resource:dbsnp,known=true,training=false,truth=false,prior=2.0 $params.dbsnp \\
+        --mode SNP \\
+        -an QD \\
+        -an FS \\
+        -an MQ
+
+    gatk -T ApplyRecalibration \\
+        -R $params.gfasta \\
+        --out ${name}_filtered_snp.vcf \\
+        --input $raw_snp \\
+        --mode SNP \\
+        --tranches_file ${name}_snp.tranches \\
+        --recal_file ${name}_snp.recal \\
+        --ts_filter_level 99.5 \\
+        -mode SNP 
+    """
+}
+
+
+/*
+ * Step 12 - Recalibrate INDELS using the Mills golden dataset 
+ * 
+*/ 
+
+process recalIndels {
+    tag "${name}"
+    publishDir "${params.outdir}/recalibratedIndels/", mode: 'copy'
```

Comment:
It may be worth _not_  saving some of these intermediate files by default if we have a merged version at the end. Can control this with a new `params.saveIntermediateVariants` or something.

Reply:
I added statements and a parameter to support these in ca0b302fe6142230afb11f35cf9fc12c52c17832